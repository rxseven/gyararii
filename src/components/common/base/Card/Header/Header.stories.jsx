import React from 'react';
import { storiesOf } from '@storybook/react';

import Header from './index';

const title = 'Components/Common/Base/Card/Header';

storiesOf(title, module).add('default', () => <Header>Card Header</Header>);
