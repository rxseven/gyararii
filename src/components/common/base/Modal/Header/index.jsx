import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired
});

class Header extends React.Component {
  handleClose = () => {
    const { onClose } = this.props;

    onClose();
  };

  render() {
    const { children } = this.props;

    return (
      <div className="modal-header">
        <h5 className="modal-title">{children}</h5>
        <button
          aria-label="Close"
          className="close"
          onClick={this.handleClose}
          type="button"
        >
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
    );
  }
}

Header.propTypes = propTypes;

export default Header;
