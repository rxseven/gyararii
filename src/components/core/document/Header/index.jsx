import React from 'react';
import styled from 'styled-components';

import Grid from 'components/common/base/Grid';

const Frame = styled.header`
  background-color: #fff;
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.2), 0 3px 3px -2px rgba(0, 0, 0, 0.14),
    0 1px 8px 0 rgba(0, 0, 0, 0.12);
  flex-shrink: 0;
  left: 0;
  padding: 0;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
  z-index: 1;
`;

const Content = styled.div`
  align-items: center;
  display: flex;
  height: 30px;
  justify-content: space-between;
  margin: 0.5rem 0;
`;

function Header() {
  return (
    <Frame>
      <Grid.Container>
        <Grid.Row>
          <Grid.Column>
            <Content>Header</Content>
          </Grid.Column>
        </Grid.Row>
      </Grid.Container>
    </Frame>
  );
}

export default Header;
