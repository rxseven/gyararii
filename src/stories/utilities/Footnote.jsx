import styled from 'styled-components';

const Footnote = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

export default Footnote;
