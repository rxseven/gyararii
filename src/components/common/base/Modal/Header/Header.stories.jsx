import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';

import notes from 'stories/shared/notes';
import Header from './index';

const title = 'Components/Common/Base/Modal/Header';
const component = <Header onClose={action('onClose')}>Modal Header</Header>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
