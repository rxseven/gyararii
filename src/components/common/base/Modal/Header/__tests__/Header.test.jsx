import React from 'react';
import { fireEvent } from 'react-testing-library';

import { factory, toMarkup } from 'tests/utilities';
import Header from '../index';

// Arrange
const seed = { content: 'content' };
const source = {
  children: <span>{seed.content}</span>,
  onClose: jest.fn()
};
const input = { ...seed, ...source };

// Setup
function setup(props) {
  return factory(Header, source, props);
}

// Test suites
describe('<Modal.Header />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed children correctly', () => {
    const expected = {
      content: input.content,
      html: input.children
    };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
    expect(component).toContainHTML(toMarkup(expected.html));
  });

  it('should call onClose() when the close button is clicked', () => {
    const expected = { called: 1 };
    const { getByText } = setup();

    fireEvent.click(getByText('Ã—'));

    expect(input.onClose).toHaveBeenCalledTimes(expected.called);
  });
});
