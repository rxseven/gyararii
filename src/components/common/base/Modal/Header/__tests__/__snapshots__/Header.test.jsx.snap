// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal.Header /> Snapshot tests should render correctly 1`] = `
<div
  className="modal-header"
>
  <h5
    className="modal-title"
  >
    content
  </h5>
  <button
    aria-label="Close"
    className="close"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
</div>
`;
